src/adapter/http/src/smart_protocol/axum_server_push_protocol.rs

```diff
@@ -140,7 +193,7 @@ impl AxumServerPushProtocolInstance {
                     Ok(create_result) => self.dataset = Some(create_result.dataset),
                     Err(err) => {
                         if let CreateDatasetError::RefCollision(err) = &err {
-                            axum_write_close_payload::<DatasetPushObjectsTransferResponse>(
+                            axum_write_payload::<DatasetPushObjectsTransferResponse>(
```

Comment:
In the pull flow it looks like we no longer write errors manually into the channel or do complex error mapping - we just use `?` and the handler in `serve` takes care of the rest.

I wonder why this didn't happen for push flow.

Reply:
Yeah, changed